sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/ForeachSinkSuite.scala

```diff
@@ -204,6 +204,55 @@ class ForeachSinkSuite extends StreamTest with SharedSQLContext with BeforeAndAf
       query.stop()
     }
   }
+
+  test("foreach with watermark: append") {
```

Comment:
@tdas As no eviction with `complete` mode, it will always output all data. So basically, the test "watermark + complete" is not super helpful.